### ISSUE TITLE
chore: pin debug keystore in distribute.sh to ensure consistent APK signing

### ISSUE THREAD

**Issue by Chitiiran** [Chitiiran/dytty#118]
## Problem

`distribute.sh` builds debug APKs with `flutter build apk --debug` but does not specify a signing keystore. It relies on the machine's default debug keystore (`~/.android/debug.keystore`), which means:

- **Different machines produce differently-signed APKs** — if you distribute from a second machine (or CI), Android treats it as a different app and refuses to install over the existing one
- **Reinstall required** — testers get "app not installed" errors and must uninstall first, losing any local data
- **Keystore loss = broken update path** — if the debug keystore is lost or regenerated, all existing installs become un-upgradeable

This likely contributed to the broken `0.1.8+10` distribution — if the APK was signed with a different key than the previously installed version, Android silently rejects the update.

## Solution

1. **Check in a shared debug keystore** (or store it as a CI secret) that all distributions use
2. **Update `distribute.sh`** to explicitly pass signing config:
   ```bash
   flutter build apk --debug \
     --dart-define=... \
     --target-platform android-arm64 \
     # signing handled by build.gradle debug signingConfig
   ```
3. **Configure `android/app/build.gradle`** debug signing to use the pinned keystore:
   ```groovy
   signingConfigs {
       debug {
           storeFile file("../debug.keystore")
           storePassword "android"
           keyAlias "androiddebugkey"
           keyPassword "android"
       }
   }
   ```
4. **Verify CI uses the same keystore** — the `DEBUG_KEYSTORE_BASE64` secret in CI should decode to the same file

## Acceptance Criteria

- [ ] A shared debug keystore is committed to the repo (or documented as a required local setup step)
- [ ] `android/app/build.gradle` debug signingConfig points to the shared keystore
- [ ] `distribute.sh` builds produce APKs signed with the shared keystore
- [ ] CI builds produce APKs signed with the same keystore
- [ ] Documented in `docs/planning/RELEASE.md`

## Success Criteria

- [ ] APK built locally and APK built in CI can update each other without reinstall
- [ ] Two consecutive `distribute.sh` runs produce APKs that install over each other cleanly

## Relates To

- #117 (release traceability)
- #95 (CI/CD auto-distribution)
